src/ln/channel.rs

```diff
@@ -2040,6 +2046,14 @@ impl Channel {
 
 	}
 
+	/// Set channel fields with remote peer forwarding parameters, used mainly by ChannelDetails to feed the Router
+	pub fn channel_update(&mut self, msg: &msgs::ChannelUpdate) {
+		self.their_to_self_delay = msg.contents.cltv_expiry_delta;
+		self.their_htlc_minimum_msat = msg.contents.htlc_minimum_msat;
```

Comment:
This should be sanity-checked against funding_satoshis (see new_from_req), and while ideally we'd sanity-check against the channel_limits we dont keep those around and I think its probably OK to let counterparties update such things if they need to temporarily disable a channel. This needs to be documented in Config, though, noting that a counterparty can increase this value once the channel is established beyond the minimum, but this is equivalent to them just rejecting all HTLCs so we can't enforce it anyway.

Reply:
Okay, added comments in Config and sanity-check against channel_value_satoshis (funding_satoshis) and if fail reject the whole update, but should we send back a ChannelError:Close here, or as you said interpret it as temporary disabling of channel?

Reply:
I think if they send back something completely bogus, nothing wrong with a ChannelError::Close.

Reply:
Updated, but don't do comparison for others fields because I think we don't have reference bases